File: backend/src/main/java/com/matheushenrique/nexum/services/UserService.java
package com.matheushenrique.nexum.services;

import com.matheushenrique.nexum.dtos.request.ChangePasswordRequest;
import com.matheushenrique.nexum.dtos.request.UpdateProfileRequest;
import com.matheushenrique.nexum.dtos.response.MessageResponse;
import com.matheushenrique.nexum.dtos.response.UserResponse;
import java.util.UUID;

public interface UserService {
    UserResponse getMe(UUID userId);
    MessageResponse updateProfile(UUID userId, UpdateProfileRequest request);
    MessageResponse changePassword(UUID userId, ChangePasswordRequest request);
}

File: frontend/src/services/userService.ts
import api from './authService'
import type { UserProfileResponse, UpdateProfileRequest, ChangePasswordRequest } from '../types/user'
import type { MessageResponse } from '../types/auth'

export const userService = {
    getMe: async (): Promise<UserProfileResponse> => {
        const response = await api.get<UserProfileResponse>('/v1/users/me')
        return response.data
    },

    updateProfile: async (data: UpdateProfileRequest): Promise<MessageResponse> => {
        const response = await api.put<MessageResponse>('/v1/users/me', data)
        return response.data
    },

    changePassword: async (data: ChangePasswordRequest): Promise<MessageResponse> => {
        const response = await api.put<MessageResponse>('/v1/users/me/password', data)
        return response.data
    },
}
